integration-tests/smoke/vrfv2plus_test.go

```diff
@@ -1364,6 +1367,211 @@ func TestVRFV2PlusWithBHS(t *testing.T) {
+		vrfv2plus.LogRandRequest(
+			l,
+			vrfv2PlusContracts.VRFV2PlusConsumer[0].Address(),
+			vrfv2PlusContracts.CoordinatorV2Plus.Address(),
+			subID,
+			isNativeBilling,
+			*configCopy.VRFv2Plus.General.MinimumConfirmations,
+			*configCopy.VRFv2Plus.General.CallbackGasLimit,
+			*configCopy.VRFv2Plus.General.NumberOfWords,
+			vrfv2PlusData.KeyHash,
+			*configCopy.VRFv2Plus.General.RandomnessRequestCountPerRequest,
+			*configCopy.VRFv2Plus.General.RandomnessRequestCountPerRequestDeviation,
+		)
+		_, err = vrfv2PlusContracts.VRFV2PlusConsumer[0].RequestRandomness(
+			vrfv2PlusData.KeyHash,
+			subID,
+			*configCopy.VRFv2Plus.General.MinimumConfirmations,
+			*configCopy.VRFv2Plus.General.CallbackGasLimit,
+			isNativeBilling,
+			*configCopy.VRFv2Plus.General.NumberOfWords,
+			*configCopy.VRFv2Plus.General.RandomnessRequestCountPerRequest,
+		)
+		require.NoError(t, err, "error requesting randomness")
+		initialReqRandomWordsRequestedEvent, err := vrfv2PlusContracts.CoordinatorV2Plus.WaitForRandomWordsRequestedEvent(
+			[][32]byte{vrfv2PlusData.KeyHash},
+			[]*big.Int{subID},
+			[]common.Address{common.HexToAddress(vrfv2PlusContracts.VRFV2PlusConsumer[0].Address())},
+			time.Minute*1,
+		)
+		require.NoError(t, err, "error waiting for initial request RandomWordsRequestedEvent")
+
+		// 3. create new request in a subscription with balance and wait for fulfilment
+		fundingAmt := big.NewFloat(100)
+		l.Info().
+			Str("Coordinator", vrfv2PlusContracts.CoordinatorV2Plus.Address()).
+			Int("Number of Subs to create", 1).
+			Msg("Creating and funding subscriptions, adding consumers")
+		fundedSubIDs, err := vrfv2plus.CreateFundSubsAndAddConsumers(
+			env,
+			fundingAmt,
+			fundingAmt,
+			linkToken,
+			vrfv2PlusContracts.CoordinatorV2Plus,
+			[]contracts.VRFv2PlusLoadTestConsumer{vrfv2PlusContracts.VRFV2PlusConsumer[1]},
+			1,
+		)
+		require.NoError(t, err, "error creating funded sub in replay test")
+		randomWordsFulfilledEvent, err := vrfv2plus.RequestRandomnessAndWaitForFulfillment(
+			vrfv2PlusContracts.VRFV2PlusConsumer[1],
+			vrfv2PlusContracts.CoordinatorV2Plus,
+			vrfv2PlusData,
+			fundedSubIDs[0],
+			isNativeBilling,
+			*configCopy.VRFv2Plus.General.MinimumConfirmations,
+			*configCopy.VRFv2Plus.General.CallbackGasLimit,
+			*configCopy.VRFv2Plus.General.NumberOfWords,
+			*configCopy.VRFv2Plus.General.RandomnessRequestCountPerRequest,
+			*configCopy.VRFv2Plus.General.RandomnessRequestCountPerRequestDeviation,
+			configCopy.VRFv2Plus.General.RandomWordsFulfilledEventTimeout.Duration,
+			l,
+		)
+		require.NoError(t, err, "error requesting randomness and waiting for fulfilment")
+		require.False(t, randomWordsFulfilledEvent.OnlyPremium, "RandomWordsFulfilled Event's `OnlyPremium` field should be false")
+		require.Equal(t, isNativeBilling, randomWordsFulfilledEvent.NativePayment, "RandomWordsFulfilled Event's `NativePayment` field should be false")
```

Comment:
can be removed, is not the purpose for this test

Reply:
Removed native payment check